pkg/controller/atlascustomrole/custom_role.go

```diff
@@ -0,0 +1,119 @@
+package atlascustomrole
+
+import (
+	"fmt"
+	"reflect"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/translation/customroles"
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/controller/workflow"
+)
+
+type roleController struct {
+	ctx       *workflow.Context
+	projectID string
+	service   customroles.CustomRoleService
+	role      *akov2.AtlasCustomRole
+	dpEnabled bool
+	k8sClient client.Client
+}
+
+func handleCustomRole(ctx *workflow.Context, k8sClient client.Client, akoCustomRole *akov2.AtlasCustomRole,
+	dpEnabled bool) workflow.Result {
+	ctx.Log.Debug("starting custom role processing")
+	defer ctx.Log.Debug("finished custom role processing")
+
+	r := roleController{
+		ctx:       ctx,
+		service:   customroles.NewCustomRoles(ctx.SdkClient.CustomDatabaseRolesApi),
+		role:      akoCustomRole,
+		dpEnabled: dpEnabled,
+		k8sClient: k8sClient,
+	}
+
+	return r.Reconcile()
+}
+
+func (r *roleController) Reconcile() workflow.Result {
+	if r.role.Spec.ProjectRef != nil {
+		project := &akov2.AtlasProject{}
+		err := r.k8sClient.Get(r.ctx.Context, *(r.role.Spec.ProjectRef.GetObject(r.role.GetNamespace())), project)
+		if err != nil {
+			return workflow.Terminate(workflow.ProjectCustomRolesReady, err.Error())
+		}
+		if project.Status.ID == "" {
+			return workflow.Terminate(workflow.ProjectCustomRolesReady,
+				fmt.Sprintf("the referenced AtlasProject resource '%s' doesn't have ID (status.ID is empty)", project.GetName()))
+		}
+		r.projectID = project.Status.ID
+	} else if r.role.Spec.ExternalProjectIDRef != nil {
+		r.projectID = r.role.Spec.ExternalProjectIDRef.ID
+	}
+
+	roleFoundInAtlas := false
+	roleInAtlas, err := r.service.Get(r.ctx.Context, r.projectID, r.role.Spec.Role.Name)
+	switch {
+	case err != nil:
+		return workflow.Terminate(workflow.ProjectCustomRolesReady, err.Error())
+	case err == nil && (roleInAtlas != customroles.CustomRole{}):
+		roleFoundInAtlas = true
+	}
```

Comment:
I think a simple if here makes things more readable.

```suggestion
	roleInAtlas, err := r.service.Get(r.ctx.Context, r.projectID, r.role.Spec.Role.Name)
	if err != nil {
		return workflow.Terminate(workflow.ProjectCustomRolesReady, err.Error())
	}
	
	roleFoundInAtlas :=  roleInAtlas != customroles.CustomRole{}
```

Reply:
Thanks! Fixed